src/app/jscaip/AI/GameNode.ts

```diff
@@ -79,49 +87,81 @@ export class GameNode<M extends Move, S extends GameState> {
      * You can view the DOT graph with a tool like xdot,
      * or by pasting it on a website like https://dreampuf.github.io/GraphvizOnline/
      */
-    public printDot(rules: AbstractRules,
-                    labelFn?: (node: GameNode<M, S>) => string,
-                    max?: number,
-                    level: number = 0,
-                    id: number = 0,
-                    config: MGPOptional<RulesConfig> = MGPOptional.empty())
-    : number
+    public showDot(rules: AbstractRules,
+                   config: MGPOptional<RulesConfig>,
+                   labelFn?: (node: GameNode<M, S>) => string,
+                   max?: number,
+                   level: number = 0,
+                   id: number = 0)
+    : { dot: string, nextId: number, winner: PlayerOrNone }
     {
+        let buffer: string = '';
         if (level === 0) {
-            console.log('digraph G {');
+            buffer += 'digraph G {\n';
         }
         const gameStatus: GameStatus = rules.getGameStatus(this, config);
-        let color: string = 'white';
+
+        let winner: PlayerOrNone = PlayerOrNone.NONE;
         if (gameStatus.isEndGame) {
-            switch (gameStatus.winner) {
-                case Player.ZERO:
-                    color = '#994d00';
-                    break;
-                case Player.ONE:
-                    color = '#ffc34d';
-                    break;
-                default:
-                    color = 'grey';
-                    break;
-            }
+            winner = gameStatus.winner;
         }
-        let label: string = `#${this.gameState.turn}: ${this.id}`;
-        if (labelFn) {
-            label += ` - ${labelFn(this)}`;
-        }
-        console.log(`    node_${id} [label="${label}", style=filled, fillcolor="${color}"];`);
 
         let nextId: number = id+1;
+        const currentPlayer: Player = this.gameState.getCurrentPlayer();
+        let onlyLosses: boolean = true;
         if (max === undefined || level < max) {
-            for (const child of this.children.getValueList()) {
-                console.log(`    node_${id} -> node_${nextId} [label="${child.previousMove.get()}"];`);
-                nextId = child.printDot(rules, labelFn, max, level+1, nextId);
+            const children: GameNode<M, S>[] = this.children.getValueList();
+            for (const child of children) {
+                const playerColor: string = this.getPlayerDotColor(this.gameState.getCurrentPlayer());
+                buffer += `    node_${id} -> node_${nextId} [label="${child.previousMove.get()}"; color="${playerColor}"];\n`;
+                const result: { dot: string, nextId: number, winner: PlayerOrNone } =
+                    child.showDot(rules, config, labelFn, max, level+1, nextId);
+                nextId = result.nextId;
+                buffer += result.dot;
+                // If current player has a guaranteed win with this move, color the node with player's color
```

Comment:
it doesn't do it correctly!

Reply:
Sooooo whadowedo ?
Car ok le ticket est pas "fix debility" mais "enhance debuggability" maiiiiis là du coup il le fait pas non plus right :D ?

Reply:
j'me suis auto-plussé pour dire que c'est fixé !

Reply:
on rajouterais peut être un truc pour configurer la depth de l'arbre :D ?

Reply:
c'est faite !